docs/internals/ARCHITECTURE.md

```diff
@@ -103,127 +98,118 @@ hash map that stores accounts keyed by their IDs. Processing a transfer is then
-is represented by an explicit struct which is statically allocated as a field of a  component that
-owns the structure. So there's always a natural limit of how many concurrent tasks can be in flight!
-
-In summary, **static allocation is not the goal, but rather a forcing function to ensure that
-everything has a limit**, and a natural consequence of the limits.
+TigerBeetle also doesn't do arena allocation. It allocates an appropriately fixed-sized arena for 
+allocations at the start and fails with an 'out of memory' error if the limit is exceeded. In 
+TigerBeetle, static memory allocation is a consequence of everything having an explicit upper bound.
+
+When TigerBeetle starts, it computes the worst case memory allocation scenario for every "object" in 
+the system. After startup, no new objects are created therefore no dynamic memory allocation 
+is needed. The upper bounds are computed using a combination of static and dynamic information.
+
+Knowing the limits ensures that the system continues to function correctly even when overloaded. For 
+example, TigerBeetle doesn't need to have _explicit_ code for handling backpressure. If everything has 
+a limit, there's nothing to grow without bound to begin with. Backpressure arsis from the entire 
+system of components needing to honor each-other's limits.
+
+Another interesting consequence of static limits is runway concurrency. In highly concurrent 
+applications there are more concurrent tasks than the number of underlying resources available. This 
+leads to oversubscription. A concurrent task is usually a closure allocated somewhere on the heap and 
+registered with an event loop (a `Box<dyn Future>`). TigerBeetle _can't_ heap allocate structures at 
+runtime. Each TigerBeetle "future" is represented by an explicit struct which is statically allocated 
+as a field of a component that owns the structure. There's always a natural limit of how many concurrent 
+tasks can be in flight!
+
+In summary, **static allocation is a forcing function to ensure that everything has a limit**, 
+and a natural consequence of these limits.
 
 Somewhat surprisingly, our experience is that static allocation also simplifies the system greatly.
-You need to spend more time up-front thinking through what data you need to store where, but, after
-the initial design, the interplay of the connected limits between the components tends to just work
-out.
+You spend more time thinking up-front but after the initial design the interplay tends to just work out.
 
-Of course static allocation makes the system faster, and greatly reduces latency variation, but this
-also is just a consequence of knowing the limits, the "physics" of the underlying system.
+Static allocation makes the system faster and greatly reduces latency variation. This is another 
+consequence of knowing the limits and "physics" of the underlying system.
```

Comment:
Here, the `just` was actually a useful word --- what I am tryin to convey here is that while "performance" is a consequence of static allocation, and it is a _good_ consequence, but it is not the _why_ behind static allocations. 

I guess

> This is another  consequence of knowing the limits and "physics" of the underlying system, but not the main reason for choosing static allocation.